### ISSUE TITLE
The shape is stretched after the font is generated

### ISSUE THREAD

**Issue by yisibl** [googlefonts/nanoemoji#230]
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="245" height="168" viewBox="0 0 245 168">
  <g fill="none">
    <path fill="#ED7A1F" d="M167.346044,84.327826 C167.346044,118.170061 146.959974,148.679972 115.693825,161.630835 C84.4276764,174.581698 48.4387826,167.423062 24.5087085,143.492988 C0.578634349,119.562914 -6.58000169,83.5740199 6.37086113,52.3078713 C19.321724,21.0417228 49.8316349,0.655652069 83.6738703,0.655652148 C105.921096,0.471358231 127.310687,9.22760131 143.042391,24.9593053 C158.774095,40.6910093 167.530338,62.0806002 167.346044,84.327826 Z"/>
    <path fill="#849EFD" d="M244.476575,94.9770119 C244.476575,135.215721 211.007706,167.695738 156.772924,167.695737 C130.218431,168.55446 105.305171,154.877976 91.7738251,132.013567 C78.2424788,109.149158 78.2424788,80.7288003 91.7738251,57.8643914 C105.305171,34.9999825 130.218431,21.3234979 156.772924,22.1822205 C211.083771,22.1822205 244.476575,54.7383027 244.476575,94.9770119 L244.476575,94.9770119 Z"/>
  </g>
</svg>
```

![image](https://user-images.githubusercontent.com/2784308/106354014-acde7080-6329-11eb-8fed-48305fe671d5.png)

![image](https://user-images.githubusercontent.com/2784308/106354029-cf708980-6329-11eb-8bbf-7c7aec54ae41.png)

**Comment by rsheeter** [comment]
@anthrotype just as you foretold non-square viewboxes have struck (`viewBox="0 0 245 168"`) :)

@yisibl definitely a bug on our side that needs fixing. Until fixed you'll get best results if your viewbox is square.

**Comment by yisibl** [comment]
That makes sense. In the technology of generating [iconfont](https://www.iconfont.cn/), I always scale the viewBox to 1024 * 1024。

**Comment by anthrotype** [comment]
I think by default we want to center the scaled viewBox region within the font UPEM region in both the x and y directions, the same as the default `preserveAspectRatio="xMidYMid meet"`. The square UPEM region is like the "viewport" where we want to place our svg while preserving aspect ratio. That's not exactly what @yisibl is doing in #248. I think there the intention was to scale both width/height by the minimum of each and let it overflow if not square.
Not sure how to reconcile both.

Right now our assumption is that emojis (nanoemoji supposedly is for building emojis) always fit a square and are monospaced. In fact we always assign them a fixed advance width.

**Comment by rsheeter** [comment]
A config option for scaling mode may make sense?

**Comment by anthrotype** [comment]
re-reading https://docs.microsoft.com/en-us/typography/opentype/spec/svg#coordinate-systems-and-glyph-metrics

in there it's clear that the UPEM square is considered the initial viewport, and the effect of viewBox (or width/height) is to scale the SVG user coordinate system -- basically the same as our current behavior which is equivalent to _not_ preserving aspect ration (preserveAspectRation="none").

Basically if you stick that non-square SVG into an OT-SVG table as is, it will be scaled like that, and by design..

I'm very tempted to say, garbage in garbage out, fix the sources.

**Comment by yisibl** [comment]
Yes, I expect the excess part to overflow. By testing and making SVG in OT fonts in FontLab, it will maintain the aspect ratio, and the excess part will overflow.

> Basically if you stick that non-square SVG into an OT-SVG table as is, it will be scaled like that, and by design..
>I'm very tempted to say, garbage in garbage out, fix the sources.

Do you mean it should not be fixed in nanoemoji?

**Comment by anthrotype** [comment]
I think for now we will only change the default scaling behavior so that it will preserve the aspect ratio, and scale the viewBox such that it fits the UPEM square and if viewbox is a non-square rectangle it is centered both vertically and horizontally.

I would like to understand your use case better before seeing if we can implement that. I think you want to fit the shortest of viewbox width or height to the UPEM square and let the rest overflow to the right hand side. Note that that means your drawing will no longer be centered in the advance width, which by default nanoemoji sets to the same width as the UPEM for all the glyphs (assumption being emoji are monospaced).

**Comment by anthrotype** [comment]
The thing is in SVG documents we don't have a standard way to encode the advance width information. It seems that we have slightly different concerns: in our case we use this tool to create an emoji font which is meant to be set in text next to other text fonts, whereas you are using it to produce "icons" that are not normally set alongside other characters, so you don't care about the width of the glyph (as long as it overflows and does not clip).

**Comment by yisibl** [comment]
This is the effect generated by nanoemoji:
![image](https://user-images.githubusercontent.com/2784308/109962658-6adc8c00-7d26-11eb-8142-c318efda2112.png)


The following is the svg font generated by FontLab, I hope the final result remains the latter.
![image](https://user-images.githubusercontent.com/2784308/109962214-e0942800-7d25-11eb-8c4c-69cbed3e121c.png)

**Comment by yisibl** [comment]
We will also use it with other texts. The main difference is that our icons are uploaded by third-party users, so more use cases need to be considered.

**Comment by anthrotype** [comment]
ok thanks for the clarification. I think I have an idea. We can have a flag that tells the compiler whether the font you wish to build is going to be monospaced or not. In our emoji case, we'll set it to true, and we will scale viewBox to UPEM and centered while preserving aspect ratio, and set all advance widths the same value for all glyphs. When monospace if False (your use case), we scale viewBox such that the shorter between height or width fits the UPEM square, and set the advance width to the (scaled) viewbox width.

**Comment by yisibl** [comment]
It sounds very good, looking forward to your PR.

[icomoon](https://icomoon.io/) also adopted the same strategy, although it does not support the generation of color fonts.

![image](https://user-images.githubusercontent.com/2784308/109964601-d6bff400-7d28-11eb-998a-f1c11b909245.png)

[icomoon.ttf.zip](https://github.com/googlefonts/nanoemoji/files/6083606/icomoon.ttf.zip)

**Comment by rsheeter** [comment]
Must we have a flag or can we infer correct behavior from the sources?

**Comment by anthrotype** [comment]
fixed by #255
